rero_ils/dojson/utils.py

```diff
@@ -399,12 +416,32 @@ def get_contribution_link(bibid, reroid, id, key):
     prod_host = 'mef.rero.ch'
     test_host = os.environ.get('RERO_ILS_MEF_HOST', 'mef.rero.ch')
     mef_url = f'https://{test_host}/api/'
-
-    match = re_identified.search(id)
+    if type(id) is str:
+        match = re_identified.search(id)
+    else:
+        match = re_identified.search(id[0])
     if match and len(match.groups()) == 2 and key[:3] in _CONTRIBUTION_TAGS:
         match_type = match.group(1).lower()
         match_value = match.group(2)
-        if match_type == 'idref':
+        match_type.replace('de-588', 'gnd')
+        # if we have a viafid, look for the contributor
+        if match_type == "viaf":
+            url = f'{mef_url}/mef/?q=viaf_pid:{match_value}'
+            response = requests_retry_session().get(url)
+            status_code = response.status_code
+            if status_code == requests.codes.ok:
+                resp = response.json()
+                try:
+                    mdata = resp['hits']['hits'][0]['metadata']
+                    if mdata['idref']['pid']:
+                        match_value = mdata['idref']['pid']
+                        match_type = 'idref'
+                    elif mdata['gnd']['pid']:
+                        match_value = mdata['idref']['pid']
+                        match_type = 'gnd'
+                except Exception as err:
```

Comment:
General exception is not a good idea in general. What is the expected exception? `KeyError`? In such case I would use and if with `get`.